includes/wikia/tasks/AsyncTaskList.class.php

```diff
@@ -300,19 +300,18 @@ public function queue( AMQPChannel $channel = null ) {
 
 		if ( $channel === null ) {
 			$exception = null;
-			$connection = $this->connection();
-			$channel = $connection->channel();
 			try {
+				$connection = $this->connection();
+				$channel = $connection->channel();
 				$channel->basic_publish( $message, '', $this->getQueue()->name() );
```

Comment:
If it fails on basic_publish you probably still should close the connection, and maybe channel too.